vertx-platform/src/main/java/org/vertx/java/platform/impl/cli/Starter.java

```diff
@@ -296,22 +297,32 @@ private void runVerticle(boolean zip, boolean module, String main, Args args) {
       instances = 1;
     }
 
-    String configFile = args.map.get("-conf");
+    String configFilePath = args.map.get("-conf");
     JsonObject conf;
 
-    if (configFile != null) {
-      try (Scanner scanner = new Scanner(new File(configFile)).useDelimiter("\\A")){
-        String sconf = scanner.next();
-        try {
-          conf = new JsonObject(sconf);
-        } catch (DecodeException e) {
-          log.error("Configuration file does not contain a valid JSON object");
-          return;
-        }
-      } catch (FileNotFoundException e) {
-        log.error("Config file " + configFile + " does not exist");
-        return;
-      }
+    if (configFilePath != null) {
+	  String configLoader = args.map.get("-confloader");
+	  String configContent;
+	  if(configLoader != null){
+		try {  
+		  configContent = loadConfig(configLoader, configFilePath);
```

Comment:
I was thinking that the ConfigurationLoader interface would return the JsonObject instead of a String.  But I can see this making sense as well.  Mirko, what do you think?